apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -145,6 +145,34 @@ export class RushCommandLineParser extends CommandLineParser {
     return this._debugParameter.value;
   }
 
+  public get isQuiet(): boolean {
+    return this._quietParameter.value;
+  }
+
+  /**
+   * Peek at the -q/--quiet mode flag, before the command line parser is executed.
+   *
+   * Used during Rush startup.
+   */
+  public static peekIsQuiet(): boolean {
+    for (let i: number = 2; i < process.argv.length; i++) {
+      const arg: string = process.argv[i];
+      if (arg === '-q' || arg === '--quiet') return true;
+    }
+    return false;
+  }
+
+  /**
+   * Utility to determine if the app should restrict writing to the console.
+   */
+  public static shouldRestrictConsoleOutput(): boolean {
+    return (
+      CommandLineHelper.isTabCompletionActionRequest(process.argv) ||
+      RushCommandLineParser.peekIsQuiet() ||
+      process.argv.indexOf('--json') !== -1
```

Comment:
Can this be included in the `peekIsQuiet` loop? Maybe rename that function.

Reply:
🤔 I was trying to keep it as a separate, useful method, but I suppose we'd really prefer that if an actual Action wants to know if quiet mode is on, they can ask the (instantiated and evaluated) RushCommandLineParser instance.

I think I'll collapse it all into the single function and make it as tight as possible.